sonar-orchestrator/src/main/java/com/sonar/orchestrator/build/GradleBuildExecutor.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Orchestrator
+ * Copyright (C) 2011-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonar.orchestrator.build;
+
+import com.sonar.orchestrator.config.Configuration;
+import com.sonar.orchestrator.util.Command;
+import com.sonar.orchestrator.util.CommandExecutor;
+import com.sonar.orchestrator.util.StreamConsumer;
+import java.io.File;
+import java.util.Map;
+import org.apache.commons.lang.SystemUtils;
+import org.slf4j.LoggerFactory;
+
+import static com.sonar.orchestrator.util.OrchestratorUtils.checkState;
+
+public class GradleBuildExecutor extends AbstractBuildExecutor<GradleBuild> {
```

Comment:
Instead of fixing to gradle wrapper you could do something similar as there is for maven, see `FileSystem.mavenHome()` and `com.sonar.orchestrator.build.MavenBuildExecutor#executeGoal` line 63.

So that it would be possible to customise location of executable and use either wrapper or OS based build system. WDYT?